packages/Radio/src/Radio.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import PropTypes from "prop-types";
+import uuid from "uuid/v4";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import CheckIcon from "@paprika/icon/lib/Check";
+import radioStyles from "./Radio.styles";
+import Group from "./components/Group";
+
+const propTypes = {
+  a11yText: PropTypes.string,
+  children: PropTypes.node,
+  isChecked: PropTypes.bool,
+  isDisabled: PropTypes.bool,
+  canDeselect: PropTypes.bool,
+  onClick: PropTypes.func.isRequired,
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  a11yText: null,
+  isChecked: false,
+  canDeselect: false,
+  children: null,
+  isDisabled: false,
+  size: ShirtSizes.MEDIUM,
+};
+
+function Radio(props) {
+  const { a11yText, children, isChecked, canDeselect, isDisabled, size, onClick, ...moreProps } = props;
+  const radioId = React.useRef(uuid()).current;
+  const inputRef = React.useRef(null);
+
+  const styleProps = {
+    hasLabel: !!children,
+    size,
+  };
+
+  const handleKeyDown = event => {
+    if (
+      // Prevent scrolling the page with a spacerbar keypress
+      event.key === " " ||
+      // Prevent submitting forms in IE/Edge with and enter keypress
+      event.key === "Enter"
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  const handleKeyUp = event => {
+    const isTriggerKey = event.key === " "; // space key
+    if (!isDisabled && isTriggerKey) {
+      onClick(event);
+    }
+  };
+
+  const inputProps = {};
+  if (a11yText) inputProps["aria-label"] = a11yText;
+  return (
+    <div data-pka-anchor="radio" css={radioStyles} {...styleProps} {...moreProps}>
+      <input
+        onChange={() => {}}
+        onClick={onClick}
+        checked={isChecked}
+        disabled={isDisabled}
+        id={radioId}
+        onKeyUp={handleKeyUp}
+        onKeyDown={handleKeyDown}
+        ref={inputRef}
+        type="radio"
+        {...inputProps}
+      />
+      <label onKeyUp={handleKeyUp} className={canDeselect ? "deselectable" : ""} htmlFor={radioId}>
+        {children}
+
+        {canDeselect ? (
+          <CheckIcon className="radio-icon" aria-hidden data-pka-anchor="radio.icon.check" />
```

Comment:
Can we export the classes from `Radio.styles.js` and apply directly instead of using `classNames`? 

Same for the `<div>` check below.

Reply:
Tried taking this a step further and making some changes as suggested but quickly find it getting a little messy and not necessarily easier to understand. I think perhaps that could do with some refactoring in a secondary Pr but take a look at the in progress branch i have - https://github.com/acl-services/paprika/commit/297f96081768df1aa28a89ec408da2af3f27902f as I was having some issues with the disabled state

Reply:
I think you're on the right track there.  Maybe once you merge this one I'll take that branch and try finesse it into a viable PR.